drivers/ksfilter/ks/connectivity.c

```diff
@@ -283,7 +283,7 @@ KspReadMediaCategory(
     }
 
     /* query the name size */
-    Status = ZwQueryValueKey(hKey, &Name, KeyValuePartialInformation, NULL, 0, &Size);
+    Status = ZwQueryValueKey(hKey, &Name, KeyValuePartialInformation, NULL, 0, 0);
```

Comment:
I don't understand the fix... `ULONG Size;` is a local variable. Yes, it's not initialized, but it's passed as a pointer to get the size...

You probably want to initialize Size to 0.

Reply:
> You probably want to initialize Size to 0.

I already tried that, but for some reason, it does not work for me. The system still fails with bugcheck. :frowning_face:

Reply:
> /* query the name size */
>    Status = ZwQueryValueKey(hKey, &Name, KeyValuePartialInformation, NULL, 0, 0);

Then how and where do you retrieve the size? :trollface: 
No wonder the "new" code will be buggier than before.

From MSDN:
> [out] Pointer to a variable that receives the size, in bytes, of the key information. If the ZwQueryValueKey routine returns STATUS_SUCCESS, callers can use the value of this variable to determine the amount of data returned. If the routine returns STATUS_BUFFER_OVERFLOW or STATUS_BUFFER_TOO_SMALL, callers can use the value of this variable to determine the size of buffer required to hold the key information.

Reply:
OK @HBelusca. You are right. This fix is indeed wrong.
I just did this according to that OSR article, but it appeared to be the different problem from that one.
I found the actual source of problem instead and made a better fix.
The buffer is too small only because the length of the string stored in the buffer is set to `0`: https://github.com/reactos/reactos/blob/889eab78ca7c0f17a2ada9ff20de5807a293cdfe/drivers/ksfilter/ks/connectivity.c#L251
`Path` string contains `MediaPath`, which is a path of the registry key to be opened, plus `GuidString`, which is a string of media category, whose size is querying, converted from its input GUID.
So I decided that its length should be appropriate.
Indeed: after setting `Path.Length` to `MediaPath.Length + GuidString.Length`, same as it done for `Path.MaximumLength`, the system is booting for me again with MS audio stack. :smiley: 
So it's not needed to keep the buffer uninitialized.
I will force-push my branch by this fix soon. Allow me some time.

Reply:
Fixed. :ok_hand: